app/js/app/services/session.js

```diff
@@ -0,0 +1,342 @@
+          iStorage.set('token', response.data.token);
+          iStorage.set('user_id', response.data.id);
+          iStorage.set('is_registration_complete', response.data.is_registration_complete);
+        }
+
+        return response;
+      });
+    },
+
+    forgotPassword: function (email) {
+      var data = {
+        email: email
+      };
+      return $http.post(serverConfig.url + '/api/secure/forgot-password', angular.element.param(data), {
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded'
+        }
+      });
+    },
+
+    logout: function () {
+      iStorage.set('token', '');
+      iStorage.set('user_id', '');
+      iStorage.set('is_registration_complete', '');
+      session.token = null;
+      session.user_id = null;
+      session.is_registration_complete = null;
+    }
+  };
+
+  session.getToken = function () {
+    return session.token;
+  };
+
+  session.token = iStorage.get('token');
+  session.user_id = iStorage.get('user_id');
+  session.is_registration_complete = iStorage.get('is_registration_complete');
+
+  if (session.token) {
+    $http.defaults.headers.common.Token = session.token;
+  }
+
+  $rootScope.$watch(session.getToken, function () {
+    $http.defaults.headers.common.Token = session.token;
+  });
+
+  return session;
+}).factory('profile', function ($resource, serverConfig, $q, iStorage, $rootScope) {
+
+  var percents = {
+    avatar: 6,
+    first_name: 6,
+    last_name: 6,
+    birth: 6,
+    address1: 6,
+    city: 6,
+    state: 6,
+    zip: 6,
+    email: 6,
+    phone: 6,
+    sex: 5,
+    race: 5,
+    employment_status: 5,
+    marital_status: 5,
+    party: 5,
+    philosophy: 5,
+    donor: 5,
+    registration: 5
+  };
+
+  var Profile = $resource(serverConfig.url + '/api/profile/:action', null, {
+    getSuggested: {
+      isArray: true,
+      method: 'POST'
+    }
+  });
+
+  var profile;
+
+  var service = {
+
+    get: function () {
+      return profile;
+    },
+
+    load: function () {
+      var deferred = $q.defer();
+      var newProfile = Profile.get({}, function () {
+        newProfile.interests = newProfile.interests || [];
+        profile = newProfile;
+        deferred.resolve(profile);
+      }, function () {
+        deferred.reject(arguments);
+      });
+      return deferred.promise;
+    },
+
+    getPercentCompleted: function () {
+      return _(profile).reduce(function (memo, item, key) {
+        return memo + ((item && percents[key]) ? percents[key] : 0);
+      }, 0);
+    },
+
+    checkRemind: function () {
+      if (service.getPercentCompleted() > 89) {
+        return;
+      }
+      var minInterval = 604800000;
+      /* WEEK */
+      var reminder = iStorage.get(profile.id + '_reminder') || {count: 0, lastTime: 0};
+      if (reminder.count >= 3 || Date.now() - reminder.lastTime < minInterval) {
+        return;
+      }
+
+      $rootScope.confirm('Your profile isn\'t finished! Fill it out now?', function (btn) {
```

Comment:
@austinpapp We will need to talk about what is defined as the profile being complete so that the user is not annoyed with this coming up over and over again despite not providing an answer to the "Race" question or something else.